### PR TITLE
Add implementation of TxnStore

### PR DESCRIPTION
**Pull request by cw9**
@xichen2020 @robskillington

### CODE REVIEW

File: kv/etcd/store.go [hunk 1]
```diff
@@ -234,6 +234,82 @@ func (c *client) History(key string, from, to int) ([]kv.Value, error) {
...
+			etcdOps[i] = clientv3.OpPut(
+				c.opts.ApplyPrefix(opSet.Key()),
+				string(value),
+				clientv3.WithPrevKV(),
+			)
+
+			// NB(cw) We need to initiate the OpResponse with a default version here
+			// because if a key was set first time, there is no way to get the key
+			// in the response from etcd
+			opResponses[i] = kv.NewOpResponse(opSet.Key(), 1)
```
Comment (xichen2020) [inline kv/etcd/store.go]: Might be better to set the version when processing the response?

File: kv/etcd/store.go [hunk 2]
```diff
@@ -234,6 +234,82 @@ func (c *client) History(key string, from, to int) ([]kv.Value, error) {
...
+		return nil, err
+	}
+	if !r.Succeeded {
+		return nil, kv.ErrConditionCheckFailed
+	}
+
+	for i := range r.Responses {
+		// NB(cw) once we start to support other OpType
+		// we need to check the response type
+		res := r.Responses[i].GetResponsePut()
```
Comment (xichen2020) [inline kv/etcd/store.go]: Can we check the response type here and return an error if it's not a put response for forward compatibility?

Comment (cw9) [inline kv/etcd/store.go]: unfortunately there is no good way to check the type, can only check if the GetResponsePut() returns nil or not

File: kv/etcd/store.go (hunk 1 above)
Comment (cw9) [inline kv/etcd/store.go]: sounds good

File: kv/etcd/store.go [hunk 3]
```diff
@@ -242,48 +284,30 @@ func (c *client) Txn(conditions []kv.Condition, ops []kv.Op) (kv.Response, error
...
-			)
-
-			// NB(cw) We need to initiate the OpResponse with a default version here
-			// because if a key was set first time, there is no way to get the key
-			// in the response from etcd
-			opResponses[i] = kv.NewOpResponse(opSet.Key(), 1)
-		default:
-			return nil, kv.ErrUnknownOpType
-		}
+		// NB(cw) We need to initiate the OpResponse with a default version here
```
Comment (xichen2020) [inline kv/etcd/store.go]: Move or remove this comment?

**Pull request merged**